docs/rfc/rfc-1.adoc

```diff
@@ -401,12 +385,99 @@ to submit multiple redemptions. After a redemption, if the wallet has under
+  the staking contract the notifier of a fraud receives.
+- `fraud_challenge_defeat_timeout`: The amount of the the wallet has to defend
+  against a fraud challenge.
+- `fraud_challenge_deposit_amount`: The amount of ETH the party challenging the
+  wallet for fraud needs to deposit.
+
+For every UTXO spent by the wallet in an incorrect way, anyone should be able
+to provide a fraud proof. Given that a Bitcoin transaction could be so large
+that proving it on Ethereum would be impossible, fraud proofs need to be
+processed with a challenge-response approach. 
+
+When a wallet unlocks a UTXO it needs to calculate a sighash and provide
+a signature over that sighash, one for each unlocked UTXO. Fraud is reported
+for a UTXO by providing a sighash along with the wallet's signature over that
+sighash. From that moment, the wallet has a certain time to defend itself
+against the challenge and prove that the UTXO was spent in an honest way.
+
+UTXO unlocked by the wallet is spent in a fraudulent way if:
+- that unlocked UTXO is a revealed deposit that was not proved as swept and
+  can not be proved as swept, or
+- that unlocked UTXO was not and is not the main wallet's UTXO.
+
+The wallet is allowed to execute only transactions that are accepted by the
+submit sweep proof, submit redemption proof, or submit funds migrated functions.
+All other transactions are considered fraud. In other words, if the wallet signed
+some UTXO, it needs to use that UTXO, and prove to the Bridge this is a valid
+usage. If none of it happens within the timeout the wallet has to defeat against
+the challenge, this is considered a fraud.
+
+The consequence of this approach is that we need to track all UTXOs spent by the
+wallet next to the main wallet's UTXO:
+
+```
+// wallet pubkey hash to the current main's UTXO hash computed
+// as keccak256(txHash | txOutputIndex | txOutputValue)
+mapping(bytes20 => bytes32) public mainUtxos;
+
+// wallet pubkey hash to the spent main UTXO hash computed
+// as keccak256(txHash | txOutputIndex)
+mapping(bytes20 => mapping(bytes32 => bool)) spentMainUTXOs
+
+```
+
+Note that this approach is stricter than just validating the public key in the
+UTXO. For example, if we were considering that a collection of deposit UTXO’s
+plus the main UTXO for the wallet unlocked using the wallet public key and
+locked under a single UTXO using the same wallet public key is not a fraud but
+a normal sweep, the Bridge would be susceptible to attacks when the malicious
+wallet steals revealed deposits by sweeping them to another UTXO but not the main
+UTXO known by the Bridge.
+
+
+==== Submitting Fraud Challenge
+
+Anyone should be able to submit a fraud challenge. At a minimum, the function
+should accept the wallet's signature and the sighash uniquely identifying UTXO
+unlocked by the wallet. The function should validate the wallet state to make
+sure it's neither closed nor terminated. It should also validate if the signature
```

Comment:
What's the difference before closed and terminated wallets?

Reply:
https://github.com/keep-network/tbtc-v2/blob/73fcd4832ad1a1cea43d9826b49f634453337f24/solidity/contracts/bridge/Bridge.sol#L145-L164